geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/formats/BinExporter.scala

```diff
@@ -11,40 +11,48 @@ package org.locationtech.geomesa.tools.export.formats
 import java.io.OutputStream
 
 import org.geotools.data.simple.SimpleFeatureCollection
-import org.locationtech.geomesa.filter.function.BinaryOutputEncoder.{EncodingOptions, LatLonAttributes, encodeFeatureCollection}
-import org.locationtech.geomesa.tools.export.BinExportParams
+import org.geotools.factory.Hints
+import org.locationtech.geomesa.filter.function.BinaryOutputEncoder
+import org.locationtech.geomesa.filter.function.BinaryOutputEncoder.{EncodingOptions, GeometryAttribute}
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+import org.opengis.feature.simple.SimpleFeatureType
 
-class BinExporter(os: OutputStream,
-                  dtgAttribute: String,
-                  idAttribute: Option[String],
-                  latAttribute: Option[String],
-                  lonAttribute: Option[String],
-                  lblAttribute: Option[String]) extends FeatureExporter {
+class BinExporter(hints: Hints, os: OutputStream) extends FeatureExporter {
```

Comment:
I just changed the entry point - instead of having a separate command with all the flags required for BIN, now you use the regular export command with --hints and pass in the bin-specific hints.